### ISSUE TITLE
Will the root partition be protected by boot into recovery mode with rw rootfs mount?

### ISSUE THREAD

**Issue by brunoais** [morbitzer/linux-luks-tpm-boot#14]
In this guide, you mention about storing the key in the rootfs partition (by default, mounted and chrooted at `/`).

From my understanding, as long as the ramfs to boot the system and the root image of the system didn't change, the whole data becomes fully available for use of whatever is running in RAM.

Linux has a recovery mode which allows booting with img mounted at `/` as rw (not a default option, btw).
Won't the recovery mode allow full unencrypted access to the attacker, including reading the encryption key at `/secret.bin` and then use that same key to decrypt the whole drive, now, outside the control of the TPM?

If so, I just succeeded that attack in near 10 minutes in one old PC borrowed to me so I'd do the test.
I was never able to boot to the OS (it wouldn't boot without modification) originally stored but I was able to get plain text access to the whole system.

Am I missing something?

**Comment by morbitzer** [comment]
In my understanding, to enter recovery mode, you need different GRUB parameters. This should mean that at least one PCR value changes, and you won't be able to retrieve the secret from the TPM. 
``/secret.bin`` is only stored in the encrypted part of the hard disk, so you should not be able to access it without the TPM. That assumes of course that the sealing works properly, which is unfortunately not always the case. 

But it may be also the case that I am missing something. Can you guide us through the steps you took to get plain text access to the system?

**Comment by brunoais** [comment]
When I tried to boot, I selected the option in the grub menu to start linux as recovery mode (with params: `rw recovery nomodeset`). Note I didn't edit the script myself. It was an option in the menu.
The PC started and I selected to drop to console. The rest was just mounting a USB and copying of the key.

Is there a possibility I got this behavior because I did it wrong?
Problem is... I don't have the borrowed PC any more.
